packages/web-client/app/components/card-pay/prepaid-card-safe/index.ts

```diff
@@ -0,0 +1,31 @@
+import Component from '@glimmer/component';
+import { inject as service } from '@ember/service';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+
+import CardCustomization from '@cardstack/web-client/services/card-customization';
+import { PrepaidCardSafe } from '@cardstack/cardpay-sdk';
+import { taskFor } from 'ember-concurrency-ts';
+import { TaskInstance } from 'ember-concurrency';
+
+interface CardPayPrepaidCardSafeComponentArgs {
+  safe: PrepaidCardSafe;
+}
+
+export default class CardPayPrepaidCardSafeComponent extends Component<CardPayPrepaidCardSafeComponentArgs> {
+  @service declare cardCustomization: CardCustomization;
+
+  @tracked fetchTask: TaskInstance<any> | undefined;
+
+  @action fetchCustomization() {
+    if (this.args.safe.customizationDID) {
+      this.fetchTask = taskFor(
+        this.cardCustomization.fetchCardCustomization
+      ).perform(this.args.safe.customizationDID);
+    }
+  }
+
+  get customization() {
+    return this.fetchTask?.value;
+  }
```

Comment:
Do we want a fallback customization here? Would the first selectable color scheme, with no pattern, work?